sonar-php-plugin/src/main/java/org/sonar/plugins/php/PhpIniSensor.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * SonarQube PHP Plugin
+ * Copyright (C) 2010-2016 SonarSource SA
+ * mailto:contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.plugins.php;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.sonar.sslr.api.RecognitionException;
+import java.util.List;
+import org.sonar.api.batch.fs.FileSystem;
+import org.sonar.api.batch.fs.InputFile;
+import org.sonar.api.batch.rule.CheckFactory;
+import org.sonar.api.batch.rule.Checks;
+import org.sonar.api.batch.sensor.Sensor;
+import org.sonar.api.batch.sensor.SensorContext;
+import org.sonar.api.batch.sensor.SensorDescriptor;
+import org.sonar.api.batch.sensor.issue.NewIssue;
+import org.sonar.api.batch.sensor.issue.NewIssueLocation;
+import org.sonar.api.rule.RuleKey;
+import org.sonar.api.utils.log.Logger;
+import org.sonar.api.utils.log.Loggers;
+import org.sonar.php.checks.CheckList;
+import org.sonar.php.ini.PhpIniCheck;
+import org.sonar.php.ini.PhpIniIssue;
+import org.sonar.php.ini.PhpIniParser;
+import org.sonar.php.ini.tree.PhpIniFile;
+
+public class PhpIniSensor implements Sensor {
+
+  private static final Logger LOG = Loggers.get(PhpIniSensor.class);
+
+  private final CheckFactory checkFactory;
+
+  public PhpIniSensor(CheckFactory checkFactory) {
+    this.checkFactory = checkFactory;
+  }
+
+  @Override
+  public void describe(SensorDescriptor descriptor) {
+    descriptor.name("Analyzer for \"php.ini\" files");
+  }
+
+  @Override
+  public void execute(SensorContext context) {
+    Checks<PhpIniCheck> checks = checkFactory
+      .<PhpIniCheck>create(CheckList.REPOSITORY_KEY)
+      .addAnnotatedChecks(CheckList.getPhpIniChecks());
+    execute(context, checks);
+  }
+
+  @VisibleForTesting
+  protected void execute(SensorContext context, Checks<PhpIniCheck> checks) {
+    PhpIniParser parser = new PhpIniParser();
+    FileSystem fs = context.fileSystem();
+
+    for (InputFile inputFile : fs.inputFiles(fs.predicates().matchesPathPattern("**/php.ini"))) {
+      PhpIniFile phpIni;
+      try {
+        phpIni = parser.parse(inputFile.file());
+      } catch (RecognitionException e) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1166)